docs/architecture/testing/integration.md

```diff
@@ -10,17 +10,23 @@ Integration tests use the built binaries for the CLI and language runtimes. We h
 
 :::{attention}
 
-By default the integration tests use the binaries for the CLI and the language runtime plugins from `$PATH`.  You can set `PULUMI_INTEGRATION_REBUILD_BINARIES=true` in your environment to automatically re-build the binaries locally to your repository and have the integration tests use them automatically.
+Integration tests should use have a TestMain which calls the `testutils.SetupPulumiBinary()` method to set an explicit path to the binaries under test to avoid reliance on the `$PATH` which can cause the wrong binary to be used in tests, resulting in incorrect test results.
```

Comment:
```suggestion
Integration tests should have a TestMain which calls the `testutils.SetupPulumiBinary()` method to set an explicit path to the binaries under test to avoid reliance on the `$PATH` which can cause the wrong binary to be used in tests, resulting in incorrect test results.
```